src/transformers/models/layoutlmv2/tokenization_layoutlmv2.py

```diff
@@ -0,0 +1,1535 @@
+            **kwargs,
+        )
+
+        # pair = bool(pair_ids is not None)
+        # len_ids = len(ids)
+        # len_pair_ids = len(pair_ids) if pair else 0
+
+        tokens = []
+        pair_tokens = []
+        token_boxes = []
+        pair_token_boxes = []
+        labels = []
+
+        if text_pair is None:
+            if word_labels is None:
+                # CASE 1: document image classification (training + inference) + CASE 2: token classification (inference)
+                for word, box in zip(text, boxes):
+                    if len(word) < 1:  # skip empty words
+                        continue
+                    word_tokens = self.tokenize(word)
+                    tokens.extend(word_tokens)
+                    token_boxes.extend([box] * len(word_tokens))
+            else:
+                # CASE 2: token classification (training)
+                for word, box, label in zip(text, boxes, word_labels):
+                    if len(word) < 1:  # skip empty words
+                        continue
+                    word_tokens = self.tokenize(word)
+                    tokens.extend(word_tokens)
+                    token_boxes.extend([box] * len(word_tokens))
+                    # Use the real label id for the first token of the word, and padding ids for the remaining tokens
+                    labels.extend([label] + [self.pad_token_label] * (len(word_tokens) - 1))
+        else:
+            # CASE 3: document visual question answering (inference)
+            # text = question
+            # text_pair = words
+            tokens = self.tokenize(text)
+            token_boxes = [self.pad_token_box for _ in range(len(tokens))]
+
+            for word, box in zip(text_pair, boxes):
+                if len(word) < 1:  # skip empty words
+                    continue
+                word_tokens = self.tokenize(word)
+                pair_tokens.extend(word_tokens)
+                pair_token_boxes.extend([box] * len(word_tokens))
+
+        # Create ids + pair_ids
+        ids = self.convert_tokens_to_ids(tokens)
+        pair_ids = self.convert_tokens_to_ids(pair_tokens) if pair_tokens else None
+
+        # Compute the total size of the returned encodings
+        pair = bool(pair_ids is not None)
+        len_ids = len(ids)
+        len_pair_ids = len(pair_ids) if pair else 0
+        total_len = len_ids + len_pair_ids + (self.num_special_tokens_to_add(pair=pair) if add_special_tokens else 0)
+
+        # Truncation: Handle max sequence length
+        overflowing_tokens = []
+        overflowing_token_boxes = []
+        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and max_length and total_len > max_length:
+            (
+                ids,
+                token_boxes,
+                pair_ids,
+                pair_token_boxes,
+                labels,
+                overflowing_tokens,
+                overflowing_token_boxes,
+                overflowing_labels,
```

Comment:
`overflowing_labels` doesn't appear to be instantiated if this conditional is bypassed.

Reply:
Thanks for spotting, will be fixed in my next push. 

However, I've noticed some weird behaviour with slow tokenizers like `BertTokenizer`. If you specify `return_overflowing_tokens=True`, it will return the overflowing tokens in reversed order. Will open an issue for that.

Reply:
@NielsRogge good catch on the reversed order. Here it appears that the tokenization implementation accumulates the overflow explicitly. Am i understanding correctly that this current implementation ALSO returns overflowing tokens in reversed order? or does it return overflow in the logical reading order, which differs in behavior from the existing slow tokenizers provided by hugging face?  

Wondering if fixing the underlying behavior in the huggingface library to achieve parity in behavior between all fast + slow tokenizers is a stretch for this PR.  Maybe instead, adjust the behavior of the slow tokenizer in this PR to match the behavior of existing slow tokenizers like Bert... and call it out in documentation until a later fix can be implemented?

Reply:
Yeah currently, the slow tokenizer for LayoutLMv2 will also return everything in reversed order, which, to me, doesn't make sense (but for now I've implemented it in the same way as other slow tokenizers). If this will be fixed, it will be in a separate PR.

Honestly, looking at it now, there's no point in using the slow tokenizer. `LayoutLMv2TokenizerFast` is much more handy: it will handle overflowing tokens by also adding special tokens, handle padding/truncation, etc. 

Suppose that you have an example that will result in > 512 tokens, like this one:

```
from transformers import LayoutLMv2TokenizerFast

tokenizer = LayoutLMv2TokenizerFast.from_pretrained("microsoft/layoutlmv2-base-uncased")

words = ['hello' for _ in range(600)]
boxes= [[1000,1000,1000,1000] for _ in range(600)]

encoding = tokenizer(words, boxes=boxes, padding='max_length', truncation=True, return_overflowing_tokens=True)
```

Then `encoding.input_ids` will be a list of length 2, encoding.bbox will also be a list of length 2, etc., and you get the `overflow_to_sample_mapping`!